src/main/server.ts

```diff
@@ -10,12 +12,32 @@ const logger = Logger.getLogger('server');
 // TODO: set the right port for your application
 const port: number = parseInt(process.env.PORT, 10) || 3100;
 
-if (app.locals.ENV === 'development') {
-  const sslDirectory = path.join(__dirname, 'resources', 'localhost-ssl');
+function buildSslOptions(): object {
+  const sslDirectory = path.join(__dirname, 'localhost-ssl');
+
+  spawnSync('mkdir ' + sslDirectory);
+
+  const keyFile = path.join(sslDirectory, 'localhost.key');
+  const certFile = path.join(sslDirectory, 'localhost.crt');
+
+  const command = 'openssl req -nodes -x509 -newkey rsa:4096 -keyout ' + keyFile + ' -out ' + certFile + ' -sha256 -days 365 -subj "/C=GB/ST=A/L=B/O=C/OU=D/CN=E"';
```

Comment:
we shouldn't be creating  this every time.

i think this would be better done as a shell script from a script in the `package.json`, a developer would then need to run something like `yarn first-time-setup` one time, 

alternatively the start:dev command could be updated to call the command so it's seamless.

General comments on the existing implementation

1. check if files exists rather than running each time
2. don't use spawn for creating directories, there's node.js apis for that
3. use string interpolation instead of concatenation
4. use node.js apis instead of `rm -rf sslDirectory`

while this may or may not work on peoples machines who are using windows we should make it easy for them to adjust the script if they need

Reply:
Thanks for your feeddback @timja . I've implemented option 2 (enhanced `yarn start:dev`).